### ISSUE TITLE
Variables only used in MINUS should not be in-scope in the query

### ISSUE THREAD

**Issue by galgonek** [ontop/ontop#447]
### Description

The specification of SPARQL 1.1 says that "use of a variable in FILTER, or in MINUS does not cause a variable to be in-scope outside of those forms". Unfortunately, this is not respected by Ontop, and it brings some unpleasant consequences. For example, if a group pattern contains two MINUS patterns that contain variables that share the same name but that are not in-scope in the group pattern, then Ontop generates a SQL query returning a wrong result set.

### Steps to Reproduce

For example, for the mapping

```
[PrefixDeclaration]
ex: http://example.org/

[MappingDeclaration] @collection [[
mappingId	map1
target		ex:{id1} ex:prop1 ex:{val1}.
source		SELECT * FROM example.table1

mappingId	map2
target		ex:{id2} ex:prop2 ex:{val2}.
source		SELECT * FROM example.table2

mappingId	map3
target		ex:{id3} ex:prop3 ex:{val3}.
source		SELECT * FROM example.table3
]]
```

the SPARQL query

```
prefix ex: <http://example.org/>

select * where {
  ?I ex:prop1 ?V.
  minus { ?I ex:prop2 ?E }
  minus { ?I ex:prop3 ?E }
}
```

is translated as

```
SELECT v1."id1" AS "id11m1", v1."val1" AS "val11m1", v2."val2" AS "val21m2"  	# incorrect projections
FROM "example"."table1" v1
 LEFT OUTER JOIN 
"example"."table2" v2 ON v1."id1" = v2."id2" 
 LEFT OUTER JOIN 
(SELECT v3."id3" AS "id11m0", 'ontop-provenance-constant' AS "v0", v3."val3" AS "val21m1"
FROM "example"."table3" v3
) v5 ON (v1."id1" = v5."id11m0" AND v2."val2" = v5."val21m1")   		# incorrect condition
WHERE (v5."v0" IS NULL AND v2."val2" IS NULL)

```
However, the query should be translated as

```
SELECT v1."id1" AS "id11m1", v1."val1" AS "val11m1"
FROM "example"."table1" v1
 LEFT OUTER JOIN 
"example"."table2" v2 ON v1."id1" = v2."id2" 
 LEFT OUTER JOIN 
"example"."table3" v3 ON v1."id1" = v3."id3" 
WHERE (v3."val3" IS NULL AND v2."val2" IS NULL)

```
### Versions

Ontop endpoint 4.1

**Comment by ghxiao** [comment]
Hi @galgonek thanks a lot for reporting this issue. This indeed looks like a bug of Ontop. We will see how to fix it.

**Comment by bcogrel** [comment]
Hi @galgonek, the issue has been partially fixed with #405 (non-projected variable renaming) and https://github.com/ontop/ontop/commit/bb4affe02311b94b69599eadc1a5e31a12952841 (stop projecting MINUS-specific variables).

Thanks a lot for the report and providing an example!

Best,
Benjamin
